asyncapi/asgardeo/dispatcher_service.bal

```diff
@@ -22,12 +53,40 @@ service class DispatcherService {
     }
 
     public function onEventNotification(websub:ContentDistributionMessage event) returns websub:Acknowledgement|error {
-        check self.matchRemoteFunc(<json>event.content);
+        EncryptedPayload payload = check event.content.cloneWithType();
+        string|error symmetricKey = check asymmetricDecrypt(java:fromString(payload.event.payloadCryptoKey), java:fromString(self.decryptionKey));
+        if symmetricKey is string {
+            string decryptedPayload = check symmetricDecrypt(java:fromString(payload.event.payload), java:fromString(symmetricKey), java:fromString(payload.event.iv));
+            check self.emitEvent(payload, decryptedPayload);
+        } else {
+            KeyData previousKey = check self.fetchPreviousDecryptionKey(self.token, self.orgHandle);
+            string symmetricKeyRetry = check asymmetricDecrypt(java:fromString(payload.event.payloadCryptoKey), java:fromString(previousKey.key));
+            string decryptedPayload = check symmetricDecrypt(java:fromString(payload.event.payload), java:fromString(symmetricKeyRetry), java:fromString(payload.event.iv));
+            check self.emitEvent(payload, decryptedPayload);
+        }
         return websub:ACKNOWLEDGEMENT;
```

Comment:
@HasithaAthukorala If we couldn't do `return websub:ACKNOWLEDGEMENT;` like this (in case of an error). Will the hub send the event again?

Reply:
If there is an `error`, `websub:SubscriberService` will return an appropriate error response to the `hub`. But, it does not necessarily mean that `hub` will try to re-send the content. That logic is up to the `hub` developer.

Reply:
In our implementation, Hub resends the content if we didn't ACK. 

Does it have any configured retry attempts? If it doesn't have a configured retry count, the Dispatcher service will face a loop event condition.

Reply:
Its 3 attempts (got to know from @ayeshLK)